turnitintooltwo_view.class.php

```diff
@@ -1424,8 +1424,13 @@ public function get_submission_inbox_row($cm, $turnitintooltwoassignment, $parts
                                             "id" => "upload_".$submission->submission_objectid."_".$partid."_".$submission->userid,
                                             'data-eula' => $eulaaccepted, 'data-user-type' => $istutor));
 
-            if (time() > $parts[$partid]->dtdue && $turnitintooltwoassignment->turnitintooltwo->allowlate == 0 && !$istutor) {
-                $upload = "&nbsp;";
+            $duedatepassed = time() > $parts[$partid]->dtdue;
+            $latesubmissionsallowed = $turnitintooltwoassignment->turnitintooltwo->allowlate;
+            $submissiondoesnotexists = !empty($submission->submission_objectid);
```

Comment:
I find this variable name a little confusing. If `$submission->submission_objectid` is not empty then a submission exists, surely? Would it not be better to name it `$submissionexists`?

And then wouldn't the logic below need to be `!submissionexists`?

Or maybe you need to use `empty()` rather than `!empty()`, if you want to keep the variable name `$submissiondoesnotexists`. But I prefer a simpler name.

I may be misunderstanding it, but worth checking.

Reply:
Yes you're right, that makes more sense.